ci/main.cue

```diff
@@ -2,98 +2,148 @@ package main
 			}
-			env: GOMODCACHE: mounts["go mod cache"].dest
-		}
 
-		_mountSourceCode: {
-			mounts: "dagger source code": {
-				contents: _source.output
-				dest:     "/usr/src/dagger"
+			_revision: bash.#Run & {
+				input:   _image.output
+				workdir: "/src"
+				mounts: source: {
+					dest:     "/src"
+					contents: _source
+				}
+
+				script: contents: #"""
+					printf "$(git rev-parse --short HEAD)" > /revision
+					"""#
+				export: files: "/revision": string
 			}
-			workdir: mounts["dagger source code"].dest
-		}
 
-		_baseImages: #Images
+			output: _revision.export.files["/revision"]
+		}
 
-		// Go build the dagger binary
-		// depends on goLint and goTest to complete successfully
-		build: bash.#Run & {
-			_mountSourceCode
-			_mountGoCache
+		build: go.#Build & {
+			source:  _source
+			package: "./cmd/dagger/"
+			os:      client.platform.os
+			arch:    client.platform.arch
 
-			input: _baseImages.goBuilder
+			ldflags: "-s -w -X go.dagger.io/dagger/version.Revision=\(version.output)"
 
 			env: {
-				GOOS:        client.platform.os
-				GOARCH:      client.platform.arch
 				CGO_ENABLED: "0"
 				// Makes sure the linter and unit tests complete before starting the build
-				"__depends_lint":  "\(goLint.exit)"
-				"__depends_tests": "\(goTest.exit)"
+				// "__depends_lint":  "\(goLint.exit)"
+				// "__depends_tests": "\(goTest.exit)"
 			}
-
-			script: contents: #"""
-				mkdir -p /build
-				git_revision=$(git rev-parse --short HEAD)
-				go build -v -o /build/dagger \
-				 -ldflags '-s -w -X go.dagger.io/dagger/version.Revision='${git_revision} \
-				 ./cmd/dagger/
-				"""#
-
-			export: directories: "/build": _
 		}
 
 		// Go unit tests
-		goTest: bash.#Run & {
-			_mountSourceCode
-			_mountGoCache
+		test: go.#Test & {
+			// container: image: _goImage.output
+			source:  _source
+			package: "./..."
 
-			input: _baseImages.goBuilder
-			script: contents: "go test -race -v ./..."
+			// FIXME: doesn't work with CGO_ENABLED=0
+			// command: flags: "-race": true
+
+			env: {
+				// FIXME: removing this complains about lack of gcc
+				CGO_ENABLED: "0"
+			}
 		}
 
-		// Go lint using golangci-lint
-		goLint: bash.#Run & {
-			_mountSourceCode
-			_mountGoCache
+		lint: {
+			go: golangci.#Lint & {
+				source:  _source
+				version: "1.45"
+			}
+
+			shell: shellcheck.#Lint & {
+				source: _source
+			}
 
-			input: _baseImages.goLinter
-			script: contents: "golangci-lint run -v --timeout 5m"
-		}
+			markdown: markdownlint.#Lint & {
+				source: _source
+				files: ["./docs", "README.md"]
+			}
 
-		// CUE lint
-		cueLint: bash.#Run & {
-			_mountSourceCode
-
-			input: _baseImages.cue
-			script: contents: #"""
-				# Format the cue code
-				find . -name '*.cue' -not -path '*/cue.mod/*' -print | time xargs -n 1 -P 8 cue fmt -s
-				# Check that all formatted files where committed
-				test -z $(git status -s . | grep -e '^ M'  | grep .cue | cut -d ' ' -f3)
-				"""#
+			cue: docker.#Build & {
```

Comment:
Maybe add a FIXME to spin this out into a `cuelangci` package like the equivalent `golangci`?

Reply:
Done